crates/sui-sdk-crypto/Cargo.toml

```diff
@@ -0,0 +1,62 @@
+[package]
+name = "sui-sdk-crypto"
+version = "0.0.0"
+authors = ["Brandon Williams <brandon@mystenlabs.com>"]
+license = "Apache-2.0"
+edition = "2021"
+publish = false
+
+[package.metadata.docs.rs]
+# To build locally:
+# RUSTDOCFLAGS="--cfg=doc_cfg -Zunstable-options --generate-link-to-definition" RUSTC_BOOTSTRAP=1 cargo doc --all-features --no-deps --open
+all-features = true
+rustdoc-args = [
+    # Enable doc_cfg showing the required features.
+    "--cfg=doc_cfg",
+
+    # Generate links to definition in rustdoc source code pages
+    # https://github.com/rust-lang/rust/pull/84176
+    "-Zunstable-options", "--generate-link-to-definition"
+]
+
+[features]
+default = []
+ed25519 = ["dep:ed25519-dalek", "dep:rand_core", "sui-sdk/hash", "sui-sdk/serde"]
+secp256r1 = ["dep:p256", "dep:rand_core", "sui-sdk/hash", "sui-sdk/serde"]
+secp256k1 = ["dep:k256", "dep:rand_core", "signature/std", "sui-sdk/hash", "sui-sdk/serde"]
+
+[dependencies]
+signature = "2.2"
+sui-sdk = { version = "0.0.0", path = "../sui-sdk", default-features = false }
+
+# RNG support
+rand_core = { version = "0.6.4", optional = true }
+
+# ed25519 support
+ed25519-dalek = { version = "2.1.1", optional = true }
```

Comment:
noting that this is a different lib than fastcrypto uses:

- https://github.com/MystenLabs/fastcrypto/blob/3366c26a746b72707572c4f3f05915e20d5c16c2/fastcrypto/src/ed25519.rs#L79
- https://github.com/penumbra-zone/ed25519-consensus 

not quite sure what the implementation differences are--but worth investigating if using for production workloads.

Reply:
given this is a "A WIP from-scratch rust sdk for the sui blockchain", I think a note to audit/verify the integrity of this dependency should suffice